engine/src/test/java/com/redhat/cloud/notifications/EmailTemplatesRendererHelper.java

```diff
@@ -0,0 +1,170 @@
+package com.redhat.cloud.notifications;
+
+import com.redhat.cloud.notifications.db.ResourceHelpers;
+import com.redhat.cloud.notifications.ingress.Action;
+import com.redhat.cloud.notifications.models.Environment;
+import com.redhat.cloud.notifications.processors.email.EmailPendo;
+import com.redhat.cloud.notifications.qute.templates.IntegrationType;
+import com.redhat.cloud.notifications.qute.templates.TemplateDefinition;
+import com.redhat.cloud.notifications.qute.templates.TemplateService;
+import com.redhat.cloud.notifications.templates.models.DailyDigestSection;
+import io.quarkus.mailer.Mail;
+import io.quarkus.mailer.Mailer;
+import io.quarkus.test.junit.mockito.InjectSpy;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.mockito.Mockito.when;
+
+public abstract class EmailTemplatesRendererHelper {
+
+    protected static final String BUNDLE_RHEL = "rhel";
+
+    protected static final String COMMON_SECURED_LABEL_CHECK = "common template for secured env";
+
+    private static final boolean SHOULD_WRITE_ON_FILE_FOR_DEBUG = true;
+    private static final boolean SHOULD_SEND_EMAIL_FOR_DEBUG = false;
+
+    private static final String SEND_EMAIL_TO = "test.email@replace.me";
+
+    @Inject
+    Mailer mailer;
+
+    @Inject
+    protected ResourceHelpers resourceHelpers;
+
+    @Inject
+    Environment environment;
+
+    @InjectSpy
+    protected TemplateService templateService;
+
+    @BeforeEach
+    protected void initData() {
+        when(templateService.isSecuredEmailTemplatesEnabled()).thenReturn(useSecuredTemplates());
+        templateService.init();
+    }
+
+    @AfterEach
+    protected void afterEach() {
+        when(templateService.isSecuredEmailTemplatesEnabled()).thenReturn(false);
+        templateService.init();
+    }
+
+    protected final Map<String, UUID> eventTypes = new HashMap<>();
```

Comment:
```suggestion
```

Variable is not used here or by any extending class.

Reply:
Thats right, thanks.